ztp/siteconfig-generator/siteConfig/siteConfig_test.go

```diff
@@ -88,21 +75,79 @@ spec:
     diskEncryption:
       type: nbde
   - clusterName: "defaults"
-    # Without further content under diskEncryptionthe type does not get populated
+    # Without further content under diskEncryption the type does not get populated
     diskEncryption:
       type:
   - clusterName: "explicit"
     diskEncryption:
       type: none
 `
 	siteConfig := SiteConfig{}
-	_ = yaml.Unmarshal([]byte(input), &siteConfig)
+	err := yaml.Unmarshal([]byte(input), &siteConfig)
+	assert.NoError(t, err)
 
-	// Validate ClusterType
+	// Validate Disk Encryption type
 	assert.Equal(t, siteConfig.Spec.Clusters[0].DiskEncryption.Type, "nbde")
 	assert.Equal(t, siteConfig.Spec.Clusters[1].DiskEncryption.Type, "none")
 	assert.Equal(t, siteConfig.Spec.Clusters[2].DiskEncryption.Type, "none")
+}
+
+func TestRoleCounters(t *testing.T) {
+	input := `
+apiVersion: ran.openshift.io/v1
```

Comment:
There is a test case where the node role not set, it should be master by default